### PR TITLE
update symfony integration

### PR DESCRIPTION
**Pull request by DavidBadura**
The important thing to understand is that you create in the extension a dependecy injection description/configuration. you must not access the services directly otherwise you freezed the container.

### CODE REVIEW

**Comment by andrewtch** [comment]
Ok, but you can't use %doctrine.default_connection% because it is just a string ("default", for example). AFAIK connections are not services, they are accessible only via Registry... or am I missing something in DI?

**Comment by DavidBadura** [comment]
Yes, this key is a parameter and only a string. Each key can be either a parameter or service. The right key for the default connection service is `doctrine.dbal.default_connection`. So `$container->get('doctrine.dbal.default_connection')` works also. Almost all services have an id to get it. You can dump in symfony2 all services with the command `app/console container:debug`.

In doctrine have all connection service following format `doctrine.dbal.%NAME%_connection` so if you have only the name of the connection you can create the right service key:

``` php
$connectionName = 'default';
$connectionId = sprintf('doctrine.dbal.%s_connection', $connectionName);
```

The question is what we want? If you want to specify the name or service?

I have add a screenshot of a `container:debug` output;

services

**Comment by DavidBadura** [comment]
the current configuration would be:

``` yml
simple_things_entity_audit:
        audit_connection: doctrine.dbal.default_connection
```

**Comment by andrewtch** [comment]
Well, David, I was PR'ing symfony / sonata some time ago, so I'm a bit aware of these params, also I'm aware of commands (at least after I've written symfony zsh completion), YET:

```
➜  intellia git:(master) app/console container:debug | grep doctri                                             
 doctrine                                             container Doctrine\Bundle\DoctrineBundle\Registry                                                    
 doctrine.dbal.connection_factory                     container stdClass                                           
 doctrine.orm.default_entity_manager                  container Doctrine\ORM\EntityManager                                                                 
 doctrine.orm.default_manager_configurator            container Doctrine\Bundle\DoctrineBundle\ManagerConfigurator                                         
 doctrine.orm.entity_manager                          n/a       alias for "doctrine.orm.default_entity_manager"                                            
 doctrine.orm.validator.unique                        container Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntityValidator                        
 doctrine.orm.validator_initializer                   container Symfony\Bridge\Doctrine\Validator\DoctrineInitializer                                      
 form.type_guesser.doctrine                           container Symfony\Bridge\Doctrine\Form\DoctrineOrmTypeGuesser                                        
 monolog.logger.doctrine                              container Symfony\Bridge\Monolog\Logger                                                              
 sensio_framework_extra.converter.doctrine.orm        container Sensio\Bundle\FrameworkExtraBundle\Request\ParamConverter\DoctrineParamConverter 
```

and composer says 

```
,
    "require": {
        "php": ">=5.3.3",
        "symfony/symfony": "~2.4",
        "doctrine/orm": "~2.2,>=2.2.3",
```

tl;dr - either my local symfony installations are kinda broken, or symfony really sets service per connection (the the question is since when / what version / and why I don't have such services although the project is using mysql and doctrine).

I suppose I'll go to sleep and then review this tomorrow.

**Comment by DavidBadura** [comment]
I have checked it in a lot of projects. In every Symfony2 (>2.3) project i have the doctrine.dbal.default_connection service. 

do you use the doctrine-bundle?
https://github.com/doctrine/DoctrineBundle/blob/master/DependencyInjection/DoctrineExtension.php#L219

good night :)

**Comment by DavidBadura** [comment]
And sorry, I didn't know you had so much to do with symfony.

**Comment by DavidBadura** [comment]
any news?

**Comment by andrewtch** [comment]
Hi, sorry, I got a bit off the grid lately. The reason why I didn't have the connection service was a complex compiler pass implementing multiple database connections... so, you were quire right about the reasons and approach. I'm merging this and let's continue on EA.
